### PR TITLE
Add Spanish locale (closes #268)

### LINKED ISSUE #268: Spanish language support

### PR DESCRIPTION
**Pull request by juanangosto**

### CODE REVIEW

File: src/locale/es/build_format_locale/index.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+var buildFormattingTokensRegExp = require('../../_lib/build_formatting_tokens_reg_exp/index.js')
+
+function buildFormatLocale () {
```
Comment (dgdavid) [inline src/locale/es/build_format_locale/index.js]: Only one thing here: in Spanish, the months, week days and seasons's names normally are [written with lower cases](http://www.rae.es/consultas/mayuscula-o-minuscula-en-los-meses-los-dias-de-la-semana-y-las-estaciones-del-ano).

**Comment by havenchyk** [comment]
@juanangosto could you please update the list of supported langs as well? https://github.com/date-fns/date-fns/blob/master/docs/i18n.md#supported-languages

File: src/locale/es/build_format_locale/index.js (hunk 1 above)
Comment (juanangosto) [inline src/locale/es/build_format_locale/index.js]: OK, I will change it

**Comment by havenchyk** [comment]
👍

**Comment by LeshaKoss** [comment]
@juanangosto Spanish language locale was released with [1.12.0](https://github.com/date-fns/date-fns/releases/tag/v1.12.0). (diff: https://github.com/date-fns/date-fns/compare/v1.11.2...v1.12.0) Thank you for your awesome contribution! 🇪🇸🇲🇽🇨🇴🇦🇷🇵🇪🇻🇪🇨🇱🇪🇨🇬🇹🇨🇺🇧🇴🇩🇴🇭🇳🇵🇾🇸🇻🇳🇮🇨🇷🇵🇷🇵🇦🇺🇾🇬🇶

**Pull request merged**